### PR TITLE
Add warning on attempt to revoke irrevokable privileges

### PR DESCRIPTION
**Pull request by rubenwardy**
Note: you should probably review commits individually

### CODE REVIEW

File: builtin/game/chatcommands.lua [hunk 1]
```diff
@@ -204,22 +204,55 @@ core.register_chatcommand("revoke", {
...
+				if def and def.give_to_singleplayer then
+					count = count + 1
+				end
+			end
+
+			if count > 0 then
+				msg = "Warning: Unable to revoke " .. count .. " privileges from singleplayer, as they are required.\n" .. msg
+			end
+		elseif revoke_name == minetest.settings:get("name") then
+			msg = "Warning: Privileges cannot be revoked from admins.\n" .. msg
```
Comment (SmallJoker) [inline builtin/game/chatcommands.lua]: Depends on the auth handler. Warning is not required IMO, as the new privileges are listed anyway.

File: builtin/game/chatcommands.lua [hunk 2]
```diff
@@ -204,22 +204,55 @@ core.register_chatcommand("revoke", {
 			end
 		end
 
+		-- Permission check for basic_privs
+		if not core.check_player_privs(name, { privs = true }) then
+			local basic_privs =
+				core.string_to_privs(core.settings:get("basic_privs") or "interact,shout")
+			for priv, _ in pairs(revoke_privs) do
+				if not basic_privs[priv] then
+					return false, "Your privileges are insufficient (basic_privs only allows you to revoke: " .. basic_privs .. ")"
```
Comment (SmallJoker) [inline builtin/game/chatcommands.lua]: Line too long and `basic_privs` is a table

File: builtin/game/chatcommands.lua [hunk 3]
```diff
@@ -204,22 +204,55 @@ core.register_chatcommand("revoke", {
...
+			local count = 0
+			for priv, _ in pairs(revoke_privs) do
+				local def = minetest.registered_privileges[priv]
+				if def and def.give_to_singleplayer then
+					count = count + 1
+				end
+			end
+
+			if count > 0 then
+				msg = "Warning: Unable to revoke " .. count .. " privileges from singleplayer, as they are required.\n" .. msg
```
Comment (SmallJoker) [inline builtin/game/chatcommands.lua]: Line too long here too

**Comment by rubenwardy** [comment]
closing due to priv changes

**Pull request closed without merging**